compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUSelectLoweringStrategy.cpp

```diff
@@ -38,22 +28,7 @@ class LLVMCPUSelectLoweringStrategyPass
           LLVMCPUSelectLoweringStrategyPass> {
 public:
   void getDependentDialects(DialectRegistry &registry) const override {
-    // TODO(qedawkins): Once TransformStrategies is deprecated, drop the
-    // unnecessary dialect registrations.
-    // clang-format off
-    registry.insert<IREE::Codegen::IREECodegenDialect,
-                    IREE::HAL::HALDialect,
-                    IREE::LinalgExt::IREELinalgExtDialect,
-                    bufferization::BufferizationDialect,
-                    linalg::LinalgDialect,
-                    LLVM::LLVMDialect,
-                    pdl::PDLDialect,
-                    pdl_interp::PDLInterpDialect,
-                    scf::SCFDialect,
-                    tensor::TensorDialect,
-                    transform::TransformDialect,
-                    vector::VectorDialect>();
-    // clang-format on
+    registry.insert<IREE::Codegen::IREECodegenDialect>();
```

Comment:
These passes are just for strategy selection, they shouldn't be producing any new IR other than `IREE::CodegenDialect` attributes. The only reason these dialects were registered before was because transform strategies can randomly do any transformation it wants.

Reply:
If my read is correct, it is fine to drop linalg and linalg_ext. Because we don't create new linalg and linalg_ext operations/attributes/types in the pass.

> For example, a convertLinalgToLoops pass would declare the SCF Dialect as dependent, but does not need to declare Linalg.

https://mlir.llvm.org/getting_started/Faq/#registered-loaded-dependent-whats-up-with-dialects-management

Also this: https://mlir.llvm.org/docs/PassManagement/#tablegen-specification